synfig-studio/src/gui/dialogs/dialog_pluginmanager.cpp

```diff
@@ -0,0 +1,290 @@
+        return std::string("");
+    }
+
+    auto execlist = pNode->find("./exec");
+    if (!execlist.empty())
+    {
+        for (const xmlpp::Node* node : pNode->find("./name") ) {
+            const xmlpp::Element* element = dynamic_cast<const xmlpp::Element*>(node);
+            std::string lang = element->get_attribute_value("lang");
+            if (lang.empty()) {
+                return element->get_child_text()->get_content();
+            }
+        }
+    }
+}
+void
+Dialog_PluginManager::on_install_plugin_button_clicked()
+{
+    // Show the dialog and wait for a user response
+    int result = plugin_file_dialog.run();
+    switch (result) {
+        case (Gtk::RESPONSE_OK):
+        { 
+            std::vector<std::string> files;
+            PluginZipStatus pluginStatus = PluginZipStatus::PLUGIN_ZIP_NO_PLUGIN_XML;
+            etl::handle<synfig::FileSystemNative> native_fs = FileSystemNative::instance();
+            etl::handle<synfig::FileContainerZip> zip_fs = new FileContainerZip();
+            std::string zip_filename = plugin_file_dialog.get_filename();
+            filesystem::Path path_to_user_plugins = synfigapp::Main::get_user_app_directory() / filesystem::Path("plugins");
+            std::string output_path ;
+            std::string plugin_metadata_file;
+            std::string plugin_name ;
+
+            if (!zip_fs->open(plugin_file_dialog.get_filename())) {
+                synfig::error("Failed to open zip file: " + zip_filename);
+                return;
+            }
+            zip_fs->directory_scan("", files);
+            for (const auto& file : files) {
+                if (file == "plugin.xml") {
+                    plugin_metadata_file = file;
+                    break;
+                } else if(zip_fs->is_directory(file)) {
+                    std::vector<std::string> child_files;
+                    if (zip_fs->directory_scan(file, child_files)) {
+                        for (auto& innerFile : child_files) {
+                            if (innerFile == "plugin.xml") {
+                                plugin_metadata_file = file + "/" + innerFile;
+                                break;
+                            }
+                        }
+                    if (!plugin_metadata_file.empty())
+                        break;
+                    }
+                }
+            }
+            plugin_name = extract_plugin_name(*zip_fs->get_read_stream(plugin_metadata_file));
+            output_path = ((path_to_user_plugins / plugin_name).add_suffix("/")).u8string();
+            if (native_fs->is_exists(output_path)) {
+                message_dialog.set_message(_("Plugin already exists. Do you want to overwrite it?"));
+                int response = message_dialog.run();
+                if (response == Gtk::RESPONSE_CANCEL) {
+                    message_dialog.close();
+                    return;
+                }
```

Comment:
I recommend you to check if it is different of OK, instead of equals to Cancel.
If user closes the dialog instead of pressing Cancel, the dialog sets a different response.